packages/regl-worldview/src/commands/Command.js

```diff
@@ -90,45 +86,38 @@ export default class Command<T: CommandProps> extends React.Component<T> {
   }
 }
 
-// When you have children as the drawProps input, it's useful to simply call makeCommand
-// which uses SimpleCommand underneath to create a new regl component. It also handles basic hitmap
-// interactions. You'll want to subclass Command if the React children is not the drawProps input or if
-// additional data processing is needed to generate the drawProps/hitmapProps.
-export class SimpleCommand<T> extends Command<Props<T>> {
-  static defaultProps = {
-    children: [],
-    getHitmapId: undefined,
-  };
-
-  getDrawProps() {
-    return this.props.children;
+function getHitmapProps(getHitmapId, children) {
+  if (!getHitmapId || !children || children.length === 0) {
+    return undefined;
   }
 
-  getHitmapProps(): ?((T & { color: Color })[]) {
-    const { getHitmapId } = this.props;
-    if (!getHitmapId || !this.props.children || this.props.children.length === 0) {
-      return undefined;
+  return children.reduce((memo, marker) => {
+    const hitmapId = getHitmapId(marker);
+    // filter out components that don't have hitmapIds
+    if (hitmapId != null) {
+      memo.push({
+        ...marker,
+        color: intToRGB(getHitmapId(marker) || 0),
+      });
     }
-
-    return this.props.children.reduce((memo, marker) => {
-      const hitmapId = getHitmapId(marker);
-      // filter out components that don't have hitmapIds
-      if (hitmapId != null) {
-        memo.push({
-          ...marker,
-          color: intToRGB(getHitmapId(marker) || 0),
-        });
-      }
-      return memo;
-    }, []);
-  }
+    return memo;
+  }, []);
 }
 
-// factory function for creating simple regl components.
-// sample usage: const Cubes = makeCommand('Cubes', rawCommand)
-export function makeCommand<T>(name: string, command: RawCommand<any>) {
-  return class BaseComponent extends SimpleCommand<T> {
-    static displayName = name;
-    static command = command;
+// Factory function for creating simple regl components.
+// Sample usage: const Cubes = makeCommand('Cubes', rawCommand)
+// When you have children as the drawProps input, it's useful to simply call makeCommand
+// which creates a new regl component. It also handles basic hitmap interactions.
+export function makeCommand<T>(name: string, command: RawCommand<T>) {
+  const cmd = (props: Props<T>) => {
+    return (
+      <Command
+        reglCommand={command}
+        drawProps={props.children}
+        hitmapProps={getHitmapProps(props.getHitmapId, props.children)}
```

Comment:
A little unfortunate to have to run this every time even if the hitmap is not going to be used. On the other hand, I would eventually like to combine the hitmap props with regular props anyway and avoid doing any `map`.

Reply:
yeah - and `getHitmapProps` is a no-op if `props.getHitmapId` is not supplied.  I could do: `hitmapProps={props.getHitmapId ? getHitmapProps(props.getHitmapId, props.children) : undefined}` to skip the function call.